object/typeconv.go

```diff
@@ -172,15 +172,16 @@ func AsBytes(obj Object) ([]byte, *Error) {
 }
 
 func AsReader(obj Object) (io.Reader, *Error) {
+	if o, ok := obj.(interface{ AsReader() (io.Reader, *Error) }); ok {
+		return o.AsReader()
+	}
```

Comment:
Interesting... I can't recall seeing an interface defined inline before. 👍

Reply:
yeah, this works, I can create more formal interfaces if you prefer that but I think this is a nice pattern.

Reply:
It's good as-is.